### ISSUE TITLE
vmagent and SIGHUP

### ISSUE THREAD

**Issue by v98765** [VictoriaMetrics/VictoriaMetrics#335]
**Describe the bug**
При получении SIGHUP новая конфигурация не загружается

**To Reproduce**
scrape.yml
```
global:
  scrape_interval: 60s
  scrape_timeout:  30s

scrape_configs:
  - job_name: 'vmagent'
    static_configs:
    - targets: ['10.9.8.7:8429']
    relabel_configs:
      - source_labels: [__address__]
        target_label: __param_target
      - source_labels: [__param_target]
        target_label: instance
      - target_label: __address__
        replacement: 10.9.8.7:8429  # This is your vmagent
```
изменить replacement: 10.9.8.7 на replacement: 127.0.0.1
```
global:
  scrape_interval: 60s
  scrape_timeout:  30s

scrape_configs:
  - job_name: 'vmagent'
    static_configs:
    - targets: ['10.9.8.7:8429']
    relabel_configs:
      - source_labels: [__address__]
        target_label: __param_target
      - source_labels: [__param_target]
        target_label: instance
      - target_label: __address__
        replacement: 127.0.0.1:8429  # This is your vmagent
```
SIGHUP vmagent'у
```
Feb 26 10:05:00 vmagent[30555]: 2020-02-26T03:05:00.809Z        info        lib/promscrape/scraper.go:86        SIGHUP received; reloading Prometheus configs from "/etc/vmagent/scrape.yml"
Feb 26 10:05:00 vmagent[30555]: 2020-02-26T03:05:00.814Z        info        lib/promscrape/scraper.go:108        stopping Prometheus scrapers
Feb 26 10:05:00 vmagent[30555]: 2020-02-26T03:05:00.815Z        info        lib/promscrape/scraper.go:149        stopped all the 2 scrapers for `file_sd_config` targets
Feb 26 10:05:00 vmagent[30555]: 2020-02-26T03:05:00.814Z        info        lib/promscrape/scraper.go:127        stopped all the 1 scrapers for `static_config` targets
Feb 26 10:05:00 vmagent[30555]: 2020-02-26T03:05:00.815Z        info        lib/promscrape/scraper.go:112        stopped Prometheus scrapers in 0.000 seconds
Feb 26 10:05:00 vmagent[30555]: 2020-02-26T03:05:00.815Z        info        lib/promscrape/scraper.go:145        starting 2 scrapers for `file_sd_config` targets
Feb 26 10:05:00 vmagent[30555]: 2020-02-26T03:05:00.815Z        info        lib/promscrape/scraper.go:123        starting 1 scrapers for `static_config` targets
```
check targets
```
# curl http://localhost:8429/targets
job="vmagent" (1/1 up)
        state=up, endpoint=http://10.9.8.7:8429/metrics?target=10.9.8.7%3A8429, labels={instance="10.9.8.7:8429", job="vmagent"}, last_scrape=0.438s ago, scrape_duration=0.003s, error=""
```
restart vmagent
```
Feb 26 10:06:45 vmagent[6809]: 2020-02-26T03:06:45.439Z        info        app/vmagent/main.go:72        started vmagent in 0.006 seconds
Feb 26 10:06:45 vmagent[6809]: 2020-02-26T03:06:45.440Z        info        lib/promscrape/scraper.go:54        reading Prometheus configs from "/etc/vmagent/scrape.yml"
Feb 26 10:06:45 vmagent[6809]: 2020-02-26T03:06:45.443Z        info        lib/promscrape/scraper.go:145        starting 2 scrapers for `file_sd_config` targets
Feb 26 10:06:45 vmagent[6809]: 2020-02-26T03:06:45.443Z        info        lib/promscrape/scraper.go:123        starting 1 scrapers for `static_config` targets
Feb 26 10:06:45 vmagent[6809]: 2020-02-26T03:06:45.440Z        info        lib/httpserver/httpserver.go:63        starting http server at http://:8429/
Feb 26 10:06:45 vmagent[6809]: 2020-02-26T03:06:45.443Z        info        lib/httpserver/httpserver.go:64        pprof handlers are exposed at http://:8429/debug/pprof/
```
check targets
```
# curl http://localhost:8429/targets
job="vmagent" (1/1 up)
        state=up, endpoint=http://127.0.0.1:8429/metrics?target=10.9.8.7%3A8429, labels={instance="10.9.8.7:8429", job="vmagent"}, last_scrape=0.639s ago, scrape_duration=0.012s, error=""
```
**Version**
The line returned when passing `--version` command line flag to binary. For example:
```
# /usr/local/bin/vmagent --version
vmagent-20200226-022202-v1.34.0-7-g7a33da8
```

**Used command-line flags**
```
flag{name="enableTCP6", value="false"} 1
flag{name="envflag.enable", value="false"} 1
flag{name="fs.disableMmap", value="false"} 1
flag{name="graphiteListenAddr", value=""} 1
flag{name="http.disableResponseCompression", value="false"} 1
flag{name="httpAuth.password", value="secret"} 1
flag{name="httpAuth.username", value=""} 1
flag{name="httpListenAddr", value=":8429"} 1
flag{name="import.maxLineLen", value="104857600"} 1
flag{name="influxListenAddr", value=""} 1
flag{name="influxMeasurementFieldSeparator", value="_"} 1
flag{name="influxSkipSingleField", value="false"} 1
flag{name="insert.maxQueueDuration", value="1m0s"} 1
flag{name="loggerFormat", value="default"} 1
flag{name="loggerLevel", value="INFO"} 1
flag{name="loggerOutput", value="stderr"} 1
flag{name="maxConcurrentInserts", value="16"} 1
flag{name="maxInsertRequestSize", value="33554432"} 1
flag{name="memory.allowedPercent", value="60"} 1
flag{name="metricsAuthKey", value="secret"} 1
flag{name="opentsdbHTTPListenAddr", value=""} 1
flag{name="opentsdbListenAddr", value=""} 1
flag{name="opentsdbhttp.maxInsertRequestSize", value="33554432"} 1
flag{name="pprofAuthKey", value="secret"} 1
flag{name="promscrape.config", value="/etc/vmagent/scrape.yml"} 1
flag{name="promscrape.disableCompression", value="false"} 1
flag{name="promscrape.fileSDCheckInterval", value="1m0s"} 1
flag{name="promscrape.maxScrapeSize", value="16777216"} 1
flag{name="remoteWrite.basicAuth.password", value="secret"} 1
flag{name="remoteWrite.basicAuth.username", value=""} 1
flag{name="remoteWrite.bearerToken", value="secret"} 1
flag{name="remoteWrite.flushInterval", value="1s"} 1
flag{name="remoteWrite.label", value=""} 1
flag{name="remoteWrite.maxBlockSize", value="33554432"} 1
flag{name="remoteWrite.queues", value="1"} 1
flag{name="remoteWrite.relabelConfig", value=""} 1
flag{name="remoteWrite.sendTimeout", value="1m0s"} 1
flag{name="remoteWrite.showURL", value="false"} 1
flag{name="remoteWrite.tlsCAFile", value=""} 1
flag{name="remoteWrite.tlsCertFile", value=""} 1
flag{name="remoteWrite.tlsInsecureSkipVerify", value="false"} 1
flag{name="remoteWrite.tlsKeyFile", value="secret"} 1
flag{name="remoteWrite.tmpDataPath", value="/var/lib/vmagent"} 1
flag{name="remoteWrite.url", value="secret"} 1
flag{name="tls", value="false"} 1
flag{name="tlsCertFile", value=""} 1
flag{name="tlsKeyFile", value="secret"} 1
flag{name="version", value="false"} 1
```

**Comment by valyala** [comment]
@v98765 , thanks for the detailed bug report!

It should be fixed in the commit b63e4464f4ffb1d313c70143b9464b241d7ae52f . This commit will be included in the next release. In the mean time you can test it by building `vmagent` from sources according to [these docs](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/app/vmagent/README.md#how-to-build-from-sources).

**Comment by valyala** [comment]
The issue has been fixed in [v1.34.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).
